Take git_protocol from gh config

dlvhdr opened the pull request.

ldelossa commented:
@dlvhdr can we avoid the large formatter changes in this PR? Its creating a big diff.

ldelossa reviewed (approved):
LGTM! Thanks for this and good idea.

ldelossa commented:
@dlvhdr approved, but theres a merge conflict

dlvhdr commented:
@ldelossa I fixed the PR a bit. After testing it, seems like I needed to not parse it gh response as JSON.
I also removed the newline returned from the command.
Now everything works.
Feel free to land, I think I don't have the permission to.

ldelossa commented on lua/litee/gh/ghcli/init.lua:
```diff
@@ -680,4 +680,14 @@ function M.get_check_runs_by_suite(suite_id, on_read)
 end
 
+function M.get_git_protocol()
+	local cmd = [[gh config get git_protocol]]
```
@dlvhdr are you noticing odd spacing?

dlvhdr commented:
@ldelossa fixed
